cpp/src/arrow/filesystem/azurefs.h

```diff
@@ -120,6 +120,7 @@ struct ARROW_EXPORT AzureOptions {
     kDefault,
     kAnonymous,
     kStorageSharedKey,
+    kSasToken,
```

Comment:
We use `CLI` not `Cli` for CLI.

```suggestion
    kSASToken,
```